src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1540 @@
+                return;
+
+            _buildingColour = dropdown::getItemArgument(itemIndex, 4);
+            self->invalidate();
+        }
+
+        // 0x0049B37F
+        static void sub_49B37F()
+        {
+            registers regs;
+            call(0x0049B37F, regs);
+        }
+
+        // 0x0049AD46
+        static void on_tool_abort(window& self, const widget_index widgetIndex)
+        {
+            sub_49B37F();
+            ui::windows::hideGridlines();
+        }
+
+        // 0x0049ABF0
+        static void on_tool_update(window& self, const widget_index widgetIndex, const int16_t x, const int16_t y)
+        {
+            registers regs;
+            regs.esi = (int32_t)&self;
+            regs.dx = widgetIndex;
+            regs.ax = x;
+            regs.bx = y;
+            call(0x0049ABF0, regs);
+        }
+
+        // 0x0049ACBD
+        static void on_tool_down(window& self, const widget_index widgetIndex, const int16_t x, const int16_t y)
+        {
+            registers regs;
+            regs.esi = (int32_t)&self;
+            regs.dx = widgetIndex;
+            regs.ax = x;
+            regs.bx = y;
+            call(0x0049ACBD, regs);
+        }
+
+        // 0x0049AB52
+        static void on_mouse_down(window* self, widget_index widgetIndex)
+        {
+            if (widgetIndex == widx::object_colour)
+            {
+                registers regs;
+                regs.edx = widgetIndex;
+                regs.esi = (int32_t)self;
+                regs.edi = (int32_t)&self->widgets[widgetIndex];
+                call(0x0049AB72, regs);
+            }
+        }
+
+        // 0x0049B2B5
+        static void updateActiveThumb(window* self)
+        {
+            uint16_t scrollHeight = 0;
+            self->call_get_scroll_size(0, 0, &scrollHeight);
+            self->scroll_areas[0].v_bottom = scrollHeight;
+
+            auto i = 0;
+            for (; i <= self->var_83C; i++)
+            {
+                if (self->row_info[i] == self->row_hover)
+                    break;
+            }
+
+            if (i >= self->var_83C)
+                i = 0;
+
+            i = (i / 5) * rowHeight;
+
+            self->scroll_areas[0].v_top = i;
+
+            ui::scrollview::update_thumbs(self, widx::scrollview);
+        }
+
+        // 0x0049AF98
+        static void on_resize(window* self)
+        {
+            self->invalidate();
+            gfx::ui_size_t minWindowSize = { self->min_width, self->min_height };
+            gfx::ui_size_t maxWindowSize = { self->max_width, self->max_height };
+            bool hasResized = self->set_size(minWindowSize, maxWindowSize);
+            if (hasResized)
+                updateActiveThumb(self);
+        }
+
+        // 0x0049AE83
+        static void get_scroll_size(ui::window* self, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+        {
+            *scrollHeight = (4 + self->var_83C) / 5;
+            if (!*scrollHeight)
```

Comment:
Could you revert this back to `*scrollHeight == 0` its with flags that I prefer `!` syntax. scrollHeight is a unit.